expertise/create_dataset.py

```diff
@@ -300,12 +305,55 @@ def retrieve_expertise(self):
 
         return expertise
 
+    def get_papers_from_group(self, submission_group):
+        client = self.openreview_client
+        group_a_members = client.get_group(submission_group).members
+        pbar = tqdm(total=len(group_a_members), desc=submission_group + ' status...')
+        publications_by_profile_id = {}
+        all_papers = []
+
+        def get_status(profile_id):
+            pbar.update(1)
+            profile = openreview.tools.get_profile(client, profile_id)
+            if profile:
+                publications = list(openreview.tools.iterget_notes(self.openreview_client, content={'authorids': profile.id}))
+                return { 'profile_id': profile_id, 'papers': publications }
+        futures = []
+        with ThreadPoolExecutor(max_workers=self.config.get('max_workers')) as executor:
+            for profile_id in group_a_members:
+                futures.append(executor.submit(get_status, profile_id))
+        pbar.close()
+
+        for future in futures:
+            result = future.result()
+            # Convert publications to json
+            papers_json_list = []
+            for paper_note in result['papers']:
+                papers_json_list.append(paper_note.to_json())
+            publications_by_profile_id[result['profile_id']] = papers_json_list
+            all_papers = all_papers + result['papers']
+
+        # Dump publications by profile id
+        with open(self.root.joinpath('publications_by_profile_id.json'), 'w') as f:
```

Comment:
is this file being stored in the job folder or somewhere else?

shouldn't be called submission.json?

Reply:
Ah so this is a mapping that is maintained only for group scoring. The reason this is created is because the expertise model expects a list of submissions with no author metadata in the `submissions.json` file. So, we have to keep track of which author wrote which paper in the `submissions.json` file to do the averaging post-processing step. This is stored in a file because the expertise model is called separately after `create_dataset`, so it cannot just remain in memory.